modules/ducktests/tests/ignitetest/tests/snapshot_test.py

```diff
@@ -0,0 +1,162 @@
+        11. Idle verify dump3.
+        12. Checking the equality of dump1 and dump3.
+        """
+        data_cfg = DataStorageConfiguration(default=DataRegionConfiguration(persistent=True))
+
+        ignite_config = IgniteConfiguration(
+            version=IgniteVersion(ignite_version),
+            data_storage=data_cfg,
+        )
+
+        service = IgniteService(self.test_context, ignite_config, num_nodes=self.NUM_NODES - 1)
+        service.start()
+
+        control_utility = ControlUtility(service, self.test_context)
+        control_utility.activate()
+
+        client_config = ignite_config._replace(client_mode=True, data_storage=None)
+
+        streamer = IgniteApplicationService(
+            self.test_context,
+            client_config,
+            java_class_name="org.apache.ignite.internal.ducktest.tests.UuidStreamerApplication",
+            params={"cacheName": "test-cache"})
+
+        load(streamer)
+
+        node = service.nodes[0]
+
+        check_idle_verify(control_utility)
+        check_validate_indexes(control_utility)
+        dump_1 = get_dump_path(control_utility, node)
+
+        self.logger.warn("Path to dump_1=" + dump_1)
+
+        data = control_utility.snapshot_create(self.SNAPSHOT_NAME)
+
+        self.logger.warn(data)
+
+        load(streamer)
+
+        dump_2 = get_dump_path(control_utility, node)
+
+        self.logger.warn("Path to dump_2=" + dump_2)
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_2}', allow_fail=True)
+        assert len(diff) != 0, diff
+
+        service.rename_db(new_db_name='old_db')
+        service.copy_snap_to_db(self.SNAPSHOT_NAME)
+
+        service.run()
+
+        control_utility.activate()
+
+        check_idle_verify(control_utility)
+        check_validate_indexes(control_utility)
+        dump_3 = get_dump_path(control_utility, node)
+
+        self.logger.warn("Path to dump_3=" + dump_2)
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_3}', allow_fail=True)
+        assert len(diff) == 0, diff
+
+    def tearDown(self):
+        """
+        Copy work directory to result.
+        """
+        self.copy_ignite_workdir()
+
+
+def get_dump_path(control_utility: ControlUtility, node):
+    """
+    Control utils idle verify dump.
+    @return: path to dump file on node.
+    """
+    data = control_utility.idle_verify_dump(node)
+    assert 'VisorIdleVerifyDumpTask successfully' in data
+    match = re.search(r'/.*.txt', data)
+    return match[0]
+
+
+def check_validate_indexes(control_utility: ControlUtility):
+    """
+    Check indexes.
+    """
+    data = control_utility.validate_indexes()
+    assert 'no issues found.' in data, data
+
+
+def check_idle_verify(control_utility: ControlUtility):
+    """
+    Check idle verify.
+    """
+    data = control_utility.idle_verify()
+    assert 'idle_verify check has finished, no conflicts have been found.' in data, data
+
+
+def load(service_load, duration: int = 60):
```

Comment:
Is it a real need to load 1 minute long?

Reply:
Perhaps less or more, i use a minute for the test

Reply:
I believe the less a test runs the better. Why do not use only 1 sec? Or can we provide a counter limit or time limit to the application itself. After counter has exceed the application just stops. And we can continue test.

Reply:
technically we can just do one put, but that would be a strange test

Reply:
Why do you think so? I think there should be whether amount data that matters (depends on config, internal implementation or known issues) or minimal data to run a test. It can be dozens or hundreds puts because it fast, but just wait 2 mins without explanation isn't an option.